### DISCUSSION TITLE
Second list displayed empty

### DISCUSSION THREAD

**Discussion by noahfraiture** [charmbracelet/bubbletea#854] [Q&A] [answered]
Hey, I'm making a tui tool with different view with bubble tea. The problem is the following : I want to display different list to let the user choose multiple settings (and skip view if he choose some settings). Anyway it's a simple configuration with a model struct having all information and a state variable that change the used update and view function. Here's a part of my code : ```
func initialModel() model {
	return model{
		listConv: list.New([]list.Item{
			item{title: "conv1", desc: "placeholder"},
			item{title: "conv2", desc: "placeholder"},
		}, list.NewDefaultDelegate(), 0, 0),
		choiceConv: nil,
		listAI: list.New([]list.Item{
			item{title: "ai1", desc: "placeholder"},
			item{title: "ai2", desc: "placeholder"},
		}, list.NewDefaultDelegate(), 0, 0),
		choiceAI: nil,
		state:    CONV,
	}}
```
```
func (m model) Update(msg tea.Msg) (tea.Model, tea.Cmd) {
	switch m.state {
	case CONV: return m.UpdateConv(msg)
	case AI: return m.UpdateAI(msg)
	default: return nil, nil
	}}
```
Same thing for the view
`func (m model) ViewConv() string {return docStyle.Render(m.listConv.View())}`
`func (m model) ViewAI() string {return docStyle.Render(m.listAI.View())}`
```
func (m model) UpdateConv(msg tea.Msg) (tea.Model, tea.Cmd) {
	switch msg := msg.(type) {
	case tea.KeyMsg:
		switch msg.Type {
		case tea.KeyEnter:
			i, ok := m.listConv.SelectedItem().(item)
			if ok {
				m.state = AI
				m.choiceConv = &i
			}}
	case tea.WindowSizeMsg:
		h, v := docStyle.GetFrameSize()
		m.listConv.SetSize(msg.Width-h, msg.Height-v)
	}
	var cmd tea.Cmd
	m.listConv, cmd = m.listConv.Update(msg)
	return m, cmd
}
```
With ai being the same except listAI, choiceAI and a different new state

The problem is the following with image.
If I tell the updateAI to change the state to CONV, the list is well display and I can alternate between a list well displayed and a list with nothing.

Is it a problem of using two list.Model ? Or is it a bug and I should create a github issue
Thank you if you have any advices

**Comment by stefanlogue** [comment, marked as answer]
You need to update the size of your second list just like you do in your first one

**Reply by noahfraiture** [reply]
Oooh thank you !!
